liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/DevMojo.java

```diff
@@ -594,18 +594,38 @@ private void addArtifacts(org.eclipse.aether.graph.DependencyNode root, List<Fil
         List<Artifact> updatedArtifacts = new ArrayList<Artifact>();
         for (Dependency dep : dependencies) {
             boolean newDependency = true;
+
+            // match dependencies based on artifactId, groupId
             for (Dependency existingDep : existingDependencies) {
-                if (dep.getArtifactId().equals(existingDep.getArtifactId())) {
-                    newDependency = false;
-                    break;
+                if (dep.getArtifactId().equals(existingDep.getArtifactId())
```

Comment:
Instead of writing these comparisons and handling nulls, it may be simpler to do Object.equals(Object, Ojbect) or StringUtils.compare(String, String).  See https://stackoverflow.com/questions/11271554/compare-two-objects-in-java-with-possible-null-values